fabric-lifecycle-events-v1/src/main/java/net/fabricmc/fabric/api/event/lifecycle/v1/CommonLifecycleEvents.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.api.event.lifecycle.v1;
+
+import net.minecraft.util.registry.DynamicRegistryManager;
+
+import net.fabricmc.fabric.api.event.Event;
+import net.fabricmc.fabric.api.event.EventFactory;
+
+public class CommonLifecycleEvents {
+	private CommonLifecycleEvents() {
+	}
+
+	/**
+	 * Called when tags are loaded or updated.
+	 */
+	public static final Event<TagsLoaded> TAGS_LOADED = EventFactory.createArrayBacked(TagsLoaded.class, callbacks -> (registries, isClient) -> {
+		for (TagsLoaded callback : callbacks) {
+			callback.tagsLoaded(registries, isClient);
+		}
+	});
+
+	public interface TagsLoaded {
+		/**
+		 * @param registries Up-to-date registries from which the tags can be retrieved.
+		 * @param isClient True if the client just received a sync packet, false if the server just (re)loaded the tags.
+		 */
+		void tagsLoaded(DynamicRegistryManager registries, boolean isClient);
```

Comment:
This should have a more method-y name. Also, `is` prefix for params goes against yarn a bit